pytools/graph.py

```diff
@@ -240,6 +243,52 @@ def __init__(self, node: NodeT) -> None:
         self.node = node
 
 
+class NodeState(Enum):
+    WHITE = 0  # Not visited yet
+    GREY = 1   # Currently visiting
+    BLACK = 2  # Done visiting
```

Comment:
Why not use descriptive names for the node state?

Reply:
If you prefer, I can rename these, but I thought white/grey/black were standard labels in DFS (see e.g. http://www.cs.cmu.edu/afs/cs/academic/class/15750-s17/ScribeNotes/lecture9.pdf)